drivers/net/telnet.c

```diff
@@ -308,6 +308,7 @@ static void telnet_check_ctrlchar(FAR struct telnet_dev_s *priv,
         }
       else
 #endif
```

Comment:
I don't understand the else here.  The else would only be the line "signo = SIGSTP;", right?  Wouldn't the "break;" then be executed unconditioinally?  Is that the behaviour you want?  Code looks wrong.

Reply:
Let me look at the source, I might be getting confused by the difference.

Reply:
yes... the diff omits the if at line 315 and makes the logic look crazy.  I am merging now.

Reply:
@patacongo I can't see that else on 309

Reply:
SIGINT has the high priority than SIGSTP, we can't break out  immediately if we find SIGSTP char because we may SIGINT char in the later. But if SIGINT isn't enabled, it is safe to break out.